app/service/base_planning_svc.py

```diff
@@ -0,0 +1,130 @@
+import copy
+import itertools
+import re
+from base64 import b64decode
```

Comment:
move this file to `utility` folder with the other base classes

Reply:
So we do want this in /utility? It does have a lot more defined functionality than the other very minimal/generalized base classes but wasnt sure.

Reply:
I think it makes the most sense there, but it's an easy change either way.  Let's put it in utility for now, and David can make the final decision.